### ISSUE TITLE
Unable to add host with underscore in subdomain to ssh know hosts

### ISSUE THREAD

**Issue by martindahlswe** [argoproj/argo-cd#3587]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a
question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**
Trying to add a host with underscore in the subdomain fails with:
FATA[0000] rpc error: code = Unknown desc = Invalid hostname in request: gitlab.tools_stuff.example.com

**To Reproduce**
$ ssh-keyscan -p 2289 gitlab.tools_stuff.example.com | ./argocd-linux-amd64 cert add-ssh --batch

**Expected behavior**

SSH known hosts entry should be accepted and added

**Screenshots**

**Version**

```shell
argocd: v1.5.4+36bade7
  BuildDate: 2020-05-05T18:59:08Z
  GitCommit: 36bade7a2d7b69d1c0b0c4d41191f792a847d61c
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: linux/amd64
argocd-server: v1.4.2+48cced9
  BuildDate: 2020-01-24T01:07:03Z
  GitCommit: 48cced9d925b5bc94f6aa9fa4a8a19b2a59e128a
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: Version: {Version:kustomize/v3.2.1 GitCommit:d89b448c745937f0cf1936162f26a5aac688f840 BuildDate:2019-09-27T00:10:52Z GoOs:linux GoArch:amd64}
  Helm Version: v2.15.2
  Kubectl Version: v1.14.0
```

**Logs**

```
time="2020-05-14T13:22:56Z" level=info msg="received unary call /certificate.CertificateService/CreateCertificate" grpc.method=CreateCertificate grpc.request.claims="{\"iat\":1589460538,\"iss\":\"argocd\",\"nbf\":1589460538,\"sub\":\"admin\"}" grpc.request.content="certificates:<metadata:<selfLink:\"\" resourceVersion:\"\" continue:\"\" > items:<serverName:\"[gitlab.tools_stuff.example.com]:2289\" certType:\"ssh\" certSubType:\"ssh-rsa\" certData:\"[redacted]" certInfo:\"\" > items:<serverName:\"[gitlab.tools_stuff.example.com]:2289\" certType:\"ssh\" certSubType:\"ssh-ed25519\" certData:\"[redacted]\" certInfo:\"\" > > " grpc.service=certificate.CertificateService grpc.start_time="2020-05-14T13:22:56Z" span.kind=server system=grpc
time="2020-05-14T13:22:56Z" level=error msg="finished unary call with code Unknown" error="Invalid hostname in request: gitlab.tools_stuff.example.com" grpc.code=Unknown grpc.method=CreateCertificate grpc.service=certificate.CertificateService grpc.start_time="2020-05-14T13:22:56Z" grpc.time_ms=0.622 span.kind=server system=grpc```

**Comment by jannfis** [comment]
Good catch, thank you for the report!

**Comment by jannfis** [comment]
While I think that underscores in hostnames should be considered bad practice, and are also not allowed by the RFCs, we should be following the "be liberal in what you accept" principle. I have submitted fix #3596 to allow underscores in host names for certificates.
